docs/topics/api/accounts.rst

```diff
@@ -269,10 +269,74 @@ This endpoint lists the add-ons in a collection, together with collector's notes
     :>json int count: The number of results for this query.
     :>json string next: The URL of the next page of results.
     :>json string previous: The URL of the previous page of results.
-    :>json array results: An array of items in this collection.
-    :>json object results[].addon: The :ref:`add-on <addon-detail-object>` for this item.
-    :>json string|object|null results[].notes: The collectors notes for this item. (See :ref:`translated fields <api-overview-translations>`).
-    :>json int results[].downloads: The downloads that occured via this collection.
+    :>json array results: An array of :ref:`items <collection-addon-detail-object>` in this collection.
+
+
+-------------------------
+Collection Add-ons Detail
+-------------------------
+
+.. _collection-addon-detail:
+
+This endpoint gets details of a single add-on in a collection, together with collector's notes.
+
+.. http:get:: /api/v3/accounts/account/(int:user_id|string:username)/collections/(string:collection_slug)/addons/(int:addon_id|string:slug|string:guid)/
+
+    .. _collection-addon-detail-object:
+
+    :>json object addon: The :ref:`add-on <addon-detail-object>` for this item.
+    :>json string|null notes: The collectors notes for this item. (See :ref:`translated fields <api-overview-translations>`).
+    :>json int downloads: The downloads that occured via this collection.
+
+
+-------------------------
+Collection Add-ons Create
+-------------------------
+
+.. _collection-addon-create:
+
+.. note::
+    This API requires :doc:`authentication <auth>` and `Collections:Edit`
+    permission to edit collections other than your own.
+
+This endpoint allows a single add-on to be added to a collection, optionally with collector's notes.
+
+.. http:post:: /api/v3/accounts/account/(int:user_id|string:username)/collections/(string:collection_slug)/addons/
+
+    :<json string addon_id: The add-on id to be added (required).
```

Comment:
There is reason behind my madness :)
I did originally use `addon` for both but due to the way a create call is handled - it first creates the serializer instance with the data; then does the save; - it fails before `validate()` has a chance to substitute the addon object.
The reviews API works because it doesn't use a real Addon serializer for the `addon` field - it's a `SerializerMethodField` that DRF doesn't try to parse into.
I guess I can hack around it by renaming the kwarg before create is called.  Or wrapping it in a `SerializerMethodField` like in `ReviewSerializer`.  TBH this (creating based on a PK rather than the whole object) seems like a problem someone should have already fixed.